control/setup_control.go

```diff
@@ -199,6 +203,8 @@ func (p *PIDLoop) tuneSinglePID(ctx context.Context, blockIndex int) error {
 	}
 
 	p.ControlLoop.MonitorTuning(ctx)
+	tunedPID := p.ControlLoop.GetPIDVals(pidIndex)
+	(*p.TunedVals)[pidIndex] = tunedPID
 
```

Comment:
this way the errors print the types correctly, along with the pid values
```suggestion
(*p.TunedVals)[pidIndex].Type = p.PIDVals[pidIndex].Type
```